hibernate-core/src/main/java/org/hibernate/loader/Loader.java

```diff
@@ -2883,8 +2883,8 @@ protected String processDistinctKeyword(
 			String sql,
 			QueryParameters parameters) {
 		if ( !parameters.isPassDistinctThrough() ) {
-			if ( sql.startsWith( SELECT_DISTINCT ) ) {
-				return SELECT + sql.substring( SELECT_DISTINCT.length() );
+			if ( sql.contains( SELECT_DISTINCT ) ) {
+				return sql.replaceAll( SELECT_DISTINCT, SELECT );
```

Comment:
Hmmm, this changes the semantic of what we do significantly as we now replace every occurrence in the query.

There's no risk at doing that?

Reply:
This option must be explicitly activated, and it's purpose is to avoid executing the DISTINCT at the SQL level. Only if the de-duplication distinct functionality is mixed with SQL-level distinct would this be an issue. Anyway, I could rework it to remove only the top-level distinct since that's the one we are interested in.

Reply:
I think it would be safer if you can find a way to do it.

Reply:
@gsmet Done and updated the PR. Now, only the first occurrence is transformed leaving the option to use ANY, ALL queries where distinct is left as-is. I also a tets for that proving that as well, so I suppose it's going to be ok to proceed with this change.